sw/airborne/modules/multi/traffic_info.h

```diff
@@ -0,0 +1,383 @@
+{
+  LLA_COPY(ti_acs[ti_acs_id[ac_id]].lla_pos_i, *lla_pos);
+  /* clear bits for all position representations and only set the new one */
+  ti_acs[ti_acs_id[ac_id]].status = (1 << AC_INFO_POS_LLA_I);
+}
+
+/** Set velocity from ENU coordinates (int).
+* @param[in] aircraft id of aircraft info to set
+* @param[in] enu_pos position in ENU (int)
+*/
+static inline void acInfoSetPositionEnu_i(uint8_t ac_id, struct EnuCoor_i *enu_pos)
+{
+  VECT3_COPY(ti_acs[ti_acs_id[ac_id]].enu_pos_i, *enu_pos);
+  /* clear bits for all position representations and only set the new one */
+  ti_acs[ti_acs_id[ac_id]].status = (1 << AC_INFO_POS_ENU_I);
+}
+
+/** Set position from UTM coordinates (float).
+* @param[in] aircraft id of aircraft info to set
+* @param[in] enu_vel velocity in ENU (float)
+*/
+static inline void acInfoSetPositionUtm_f(uint8_t ac_id, struct UtmCoor_f *utm_pos)
+{
+  UTM_COPY(ti_acs[ti_acs_id[ac_id]].utm_pos_f, *utm_pos);
+  /* clear bits for all position representations and only set the new one */
+  ti_acs[ti_acs_id[ac_id]].status = (1 << AC_INFO_POS_UTM_F);
+}
+
+/** Set position from LLA coordinates (float).
+* @param[in] aircraft id of aircraft info to set
+* @param[in] enu_vel velocity in ENU (float)
+*/
+static inline void acInfoSetPositionLla_f(uint8_t ac_id, struct LlaCoor_f *lla_pos)
+{
+  LLA_COPY(ti_acs[ti_acs_id[ac_id]].lla_pos_i, *lla_pos);
+  /* clear bits for all position representations and only set the new one */
+  ti_acs[ti_acs_id[ac_id]].status = (1 << AC_INFO_POS_LLA_F);
+}
+
+/** Set velocity from ENU coordinates (float).
+* @param[in] aircraft id of aircraft info to set
+* @param[in] enu_pos position in ENU (float)
+*/
+static inline void acInfoSetPositionEnu_f(uint8_t ac_id, struct EnuCoor_f *enu_pos)
+{
+  VECT3_COPY(ti_acs[ti_acs_id[ac_id]].enu_pos_f, *enu_pos);
+  /* clear bits for all position representations and only set the new one */
+  ti_acs[ti_acs_id[ac_id]].status = (1 << AC_INFO_POS_ENU_F);
+}
+
+/** Set velocity from ENU coordinates (int).
+* @param[in] aircraft id of aircraft info to set
+* @param[in] enu_vel velocity in ENU (int)
+*/
+static inline void acInfoSetVelocityEnu_i(uint8_t ac_id, struct EnuCoor_i *enu_vel)
+{
+  VECT3_COPY(ti_acs[ti_acs_id[ac_id]].enu_vel_i, *enu_vel);
+  /* clear bits for all position representations and only set the new one */
+  ti_acs[ti_acs_id[ac_id]].status = (1 << AC_INFO_VEL_ENU_I);
+}
+
+/** Set velocity from ENU coordinates (float).
+ * @param[in] aircraft id of aircraft info to set
+ * @param[in] enu_vel velocity in ENU (float)
+ */
+static inline void acInfoSetVelocityEnu_f(uint8_t ac_id, struct EnuCoor_f *enu_vel)
+{
+  VECT3_COPY(ti_acs[ti_acs_id[ac_id]].enu_vel_i, *enu_vel);
+  /* clear bits for all position representations and only set the new one */
+  ti_acs[ti_acs_id[ac_id]].status = (1 << AC_INFO_VEL_ENU_F);
+}
+
+/************************ Get functions ****************************/
+
+extern void acInfoCalcPositionUtm_i(uint8_t ac_id);
```

Comment:
For the set functions, I totally forgot about that. I will update that.

For the get functions, from the current use cases, the position of the other aircraft is used to get a delta which is used to do something so I could return the position of the ownship when the requested ac_id isn't initialised. Velocity is usually used to extrapolate the other ac position with the itow difference so there I would just return zeros. With the current implementation you are likely to get really strange values if the ac isn't initialised yet. Any other ideas?

Now I write this it may also be interesting to have helper functions that just return the relative position of the other vehicle (so do the subtraction for you) cause it is almost always the goal. I was really hoping to be done with this :(